service/src/main/java/uk/nhs/adaptors/oneoneone/cda/report/mapper/CompositionMapper.java

```diff
@@ -0,0 +1,101 @@
+import uk.nhs.connect.iucds.cda.ucr.POCDMT000002UK01Component5;
+import uk.nhs.connect.iucds.cda.ucr.POCDMT000002UK01Section;
+
+import java.util.Date;
+
+import static org.hl7.fhir.dstu3.model.IdType.newRandomUuid;
+
+
+@Component
+@AllArgsConstructor
+public class CompositionMapper {
+
+    private static final String SNOMED = "371531000";
+    private static final String COMPOSITION_TITLE = "111 Report";
+    private final AuthorMapper authorMapper;
+
+    public Composition mapComposition(POCDMT000002UK01ClinicalDocument1 clinicalDocument, Encounter encounter) {
+
+        Composition composition = new Composition();
+        composition.setIdElement(newRandomUuid());
+
+        if (encounter.getEpisodeOfCareFirstRep().getResource() != null){
+            EpisodeOfCare episodeOfCare = (EpisodeOfCare) encounter.getEpisodeOfCareFirstRep().getResource();
+            composition
+                    .setCustodian(episodeOfCare.getManagingOrganization())
+                    .setCustodianTarget(episodeOfCare.getManagingOrganizationTarget());
+        }
+
+        Identifier docIdentifier = new Identifier();
+        docIdentifier.setUse(Identifier.IdentifierUse.USUAL);
+        docIdentifier.setValue(clinicalDocument.getSetId().getRoot());
+
+        composition
+                .setTitle(COMPOSITION_TITLE)
+                .setType(new CodeableConcept()
+                        .setText(SNOMED))
+                .setStatus(Composition.CompositionStatus.FINAL)
+                .setConfidentiality(Composition.DocumentConfidentiality.valueOf(clinicalDocument.getConfidentialityCode().getCode()))
+                .setEncounter(new Reference(encounter))
+                .setEncounterTarget(encounter)
+                .setSubject(encounter.getSubject())
+                .setSubjectTarget(encounter.getSubjectTarget())
+                .setDate(new Date())
+                .setIdentifier(docIdentifier);
+
+        if(clinicalDocument.getRelatedDocumentArray(0).getParentDocument().getIdArray(0).getRoot() != null){
+            Identifier relatedDocIdentifier = new Identifier();
+            relatedDocIdentifier.setUse(Identifier.IdentifierUse.USUAL);
+            relatedDocIdentifier.setValue(clinicalDocument.getRelatedDocumentArray(0).getParentDocument().getIdArray(0).getRoot());
+            composition.addRelatesTo()
+                    .setCode(Composition.DocumentRelationshipType.REPLACES)
+                    .setTarget(relatedDocIdentifier);
+        }
+
+
+        if (clinicalDocument.getAuthorArray() != null) {
+            for (POCDMT000002UK01Author author : clinicalDocument.getAuthorArray()) {
+                Encounter.EncounterParticipantComponent authorComponent = (authorMapper.mapAuthorIntoParticipantComponent(author));
+                composition.addAuthor(authorComponent.getIndividual());
+            }
+        }
+
+        if (clinicalDocument.getComponent().getStructuredBody().getComponentArray() != null) {
```

Comment:
check: 
* isSetStructuredBody()
* sizeOfComponentArray() > 0 instead of the null comparison